server/src/main/java/org/spine3/server/aggregate/AggregatePart.java

```diff
@@ -50,10 +50,26 @@
  */
 public abstract class AggregatePart<I, S extends Message, B extends Message.Builder> extends Aggregate<I, S, B> {
 
+    private final AggregateRoot<I> root;
+
     /**
      * {@inheritDoc}
      */
-    protected AggregatePart(I id) {
+    protected AggregatePart(I id, AggregateRoot<I> root) {
         super(id);
+        this.root = root;
+    }
+
+    /**
+     * Obtains a part state by its class.
+     *
+     * @param partStateClass the class of the state of the part
+     * @param <P>            the type of the part state
+     * @return the state of the part or a default state if the state was not found
+     */
+    protected <P extends Message, A extends AggregatePart<I, P, ?>> P getPartState(
```

Comment:
Do you expect any exceptions in case there is no a part with the given class, i.e. `Class<P>`? I think you should.